File: src/server/TypeLibrary.Api/Blocks/TerminalTypeReferenceView.cs
using TypeLibrary.Api.Terminals;
using TypeLibrary.Core.Common;
using TypeLibrary.Core.Terminals;

namespace TypeLibrary.Api.Blocks;

public class TerminalTypeReferenceView : HasCardinality
{
    public Direction Direction { get; set; }
    public required TerminalView Terminal { get; set; }
}

File: src/client/src/common/types/blocks/terminalTypeReferenceView.ts
import { HasCardinality } from "../common/hasCardinality";
import { Direction } from "../terminals/direction";
import { TerminalView } from "../terminals/terminalView";

export interface TerminalTypeReferenceView extends HasCardinality {
  direction: Direction;
  terminal: TerminalView;
  minCount: number;
  maxCount?: number;
}
